src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -1433,6 +1435,37 @@ private async Task WriteStreamPropertiesAsync(SelectExpandNode selectExpandNode,
             }
         }
 
+        private IEnumerable<KeyValuePair<IEdmNavigationProperty, Type>> GetNavigationPropertiesForDeepInsert(SelectExpandNode selectExpandNode, ResourceContext resourceContext)
```

Comment:
>GetNavigationPropertiesForDeepInsert [](http://example.com/codeflow?start=72&length=36)

Rather than introduce a new GetNavigationPropertiesForDeepInsert (and WriteNavigationPropertiesForDeepInsert[Async]), can we just have the existing GetNavigationPropertiesToWrite understand the deep insert scenario and return the appropriate properties to write there?

Reply:
Refactored code to use the `GetNavigationPropertiesToWrite` method.